components/style/properties.mako.rs

```diff
@@ -4783,6 +4852,92 @@ pub mod longhands {
         pub use properties::longhands::transition_duration::{get_initial_single_value};
         pub use properties::longhands::transition_duration::{get_initial_value, parse, parse_one};
     </%self:longhand>
+
+    // CSS Flexible Box Layout Module Level 1
+    // http://www.w3.org/TR/css3-flexbox/
+
+    ${new_style_struct("Flex", is_inherited=False)}
+
+    // Flex container properties
+    ${single_keyword("flex-direction", "row row-reverse column column-reverse")}
+    ${single_keyword("flex-wrap", "nowrap wrap wrap-reverse")}
+    ${single_keyword("justify-content", "flex-start flex-end center space-between space-around")}
+    ${single_keyword("align-items", "stretch flex-start flex-end center baseline")}
+    ${single_keyword("align-self", "auto flex-start flex-end center baseline stretch")}
+    ${single_keyword("align-content", "stretch flex-start flex-end center space-between space-around")}
+
+    // Flex item properties
+    ${integer_type_or_auto("order", "i32", "Specified(0)")}
+    ${integer_type_or_auto("flex-grow", "u32", "Specified(0)", "value >= 0")}
+    ${integer_type_or_auto("flex-shrink", "u32", "Specified(1)", "value >= 0")}
```

Comment:
Right now, almost none of these are supported (with the exception of `flex-direction`). However, I do intend to add support for most of them when I can. Should I remove them, or somehow feature gate them (besides using `experimental=True`)?

Also, I just realized that `align-self` is not in the right location in this list, it should be with the other flex item properties.